torchtune/data/_utils.py

```diff
@@ -33,43 +31,30 @@ def truncate(
     return tokens_truncated
 
 
-def validate_messages(
-    messages: List[Message],
-) -> None:
+def split_text_by_image_tag(content: str, image_tag: str) -> List[Dict[str, str]]:
     """
-    Given a list of messages, ensure that messages form a valid
-    back-and-forth conversation. An error will be raised if:
-
-    - There is a system message that's not the first message
-    - There are two consecutive user messages
-    - An assistant message comes before the first user message
-    - The message is empty
-    - Messages are shorter than length of 2 (min. one user-assistant turn)
-
+    Given a raw text string, split by the specified ``image_tag``
+    and form into list of dictionaries to be used in the ``Message`` content
+    field.
 
     Args:
-        messages (List[Message]): the messages to validate.
+        content (str): raw message text
+        image_tag (str): string to split the text by
 
-    Raises:
-        ValueError: If the messages are invalid.
-    """
-    if len(messages) < 2:
-        raise ValueError(
-            f"Messages must be at least length 2, but got {len(messages)} messages"
-        )
+    Returns:
+        List[Dict[str, str]]: list of dictionaries to be used in the ``Message`` content field
 
-    last_turn = "assistant"
-    for i, message in enumerate(messages):
-        if message.role == "assistant" and last_turn != "user":
-            raise ValueError(
-                f"Assistant message before expected user message at index {i} in messages"
-            )
-        if message.role == "user" and last_turn == "user":
-            raise ValueError(
-                f"Two consecutive user messages at index {i} and {i - 1} in messages"
-            )
-        if message.role == "system" and i > 0:
-            raise ValueError(
-                f"System message at index {i} in messages, but system messages must come first"
-            )
-        last_turn = message.role
+    Example:
+        >>> content = split_text_by_image_tag("<image>hello <image>world", "<image>")
+        >>> print(content)
+        [{"type": "image"}, {"type": "text", "content": "hello "}, {"type": "image"}, {"type": "text", "content": "world"}]
+    """
+    split_content = content.split(image_tag)
+    final_content_list = []
+    for i, substr in enumerate(split_content):
+        if len(substr) > 0:
+            final_content_list.append({"type": "text", "content": substr})
+        if i < len(split_content) - 1:
+            final_content_list.append({"type": "image"})
```

Comment:
So this will not support multiple consecutive image tags, right? If so that should be explicitly stated somewhere

Reply:
hmmm.... it should. let me revisit